imports/api/transactions/statement-entries/methods.js

```diff
@@ -201,18 +201,34 @@ export const recognize = new ValidatedMethod({
         Log.debug('Matching bill:', matchingBill);
       }
     }
+    let partner;
+    const entryName = entry.nameOrType();
+    if (entryName) {
+      Log.debug('Looking for partner', entryName, 'in', entry.communityId);
+      const recognizedName = Recognitions.get('name', entryName, { communityId }) || entryName;
+      partner = Partners.findOne({ communityId: entry.communityId, 'idCard.name': recognizedName });
+    } else {
+      Log.debug('No partner on statement');
+    }
+    let contraBankAccount;
+    if (!partner) {
+      const recognizedContraBAN = (entry.contraBAN && Recognitions.get('BAN', entry.contraBAN, { communityId })) || entry.contraBAN;
+      contraBankAccount = recognizedContraBAN && Accounts.findOne({ communityId, category: 'bank', BAN: recognizedContraBAN });
+    }
     // ---------------------------
     // 0th grade - 'direct' match: We find an existing payment tx, which can be mathched to this entry
     // ---------------------------
-    if (community.settings.paymentsWoStatement) {
-      const matchingTxs = Transactions.find({ communityId, category: 'payment', valueDate: entry.valueDate, amount: Math.abs(entry.amount) }).fetch();
+    if (community.settings.paymentsWoStatement && !contraBankAccount) {
+      const matchingTxs = Transactions.find({ communityId, category: 'payment', valueDate: entry.valueDate, amount: Math.abs(entry.amount) })
+        .fetch().filter(t => !t.isReconciled());
       if (matchingTxs) {
```

Comment:
Most látom hogy ezt anno rosszul irtam, mert matchingTxs az mindig defined lesz, hiszen az ures tomb is defined.
Ide az if-be szerintem matchingTxs?.length lenne a jo.

Reply:
Ajaj, miért nem volt erre teszt?!

Reply:
Ha jol latom funkcionalitast nem tori el, mert csak mindig bemegyunk ebbe az agba, akkor is ha feleslegesen, de ott kart nem okozunk, mert az ures tombben nem talalunk semmi match-et.
Tesztet igy nem tud eltorni, csak lassul a mukodes

Reply:
De, csináltam olyan tesztet, amit el tud törni. :) Mert a paymentWoStatement beállításnál is lehet olyan, hogy az entryből akarom létrehozni a paymentet, nem? Tehát akkor, ha nem talál megfelelő paymentet, menjen tovább, és ajánljon létrehozandó tx-et.